platform.js

```diff
@@ -99,26 +99,35 @@ class App {
   }
 
   copyTemplate () {
-    return fs.copy(this.opts.dir, this.originalResourcesAppDir, {
-      filter: ignore.userIgnoreFilter(this.opts),
-      dereference: this.opts.derefSymlinks
-    }).then(() => hooks.promisifyHooks(this.opts.afterCopy, [
-      this.originalResourcesAppDir,
-      this.opts.electronVersion,
-      this.opts.platform,
-      this.opts.arch
-    ])).then(() => {
-      if (this.opts.prune) {
-        return hooks.promisifyHooks(this.opts.afterPrune, [
-          this.originalResourcesAppDir,
-          this.opts.electronVersion,
-          this.opts.platform,
-          this.opts.arch
-        ])
-      } else {
-        return true
-      }
-    })
+    const asarOptions = common.createAsarOpts(this.opts) || {}
+
+    // don't copy the app if the asar filename is specified.
+    const stage = (asarOptions.filename)
+      ? Promise.resolve()
+      : fs.copy(this.opts.dir, this.originalResourcesAppDir, {
+        filter: ignore.userIgnoreFilter(this.opts),
+        dereference: this.opts.derefSymlinks
+      })
+
+    return stage
+      .then(() => hooks.promisifyHooks(this.opts.afterCopy, [
+        this.originalResourcesAppDir,
+        this.opts.electronVersion,
+        this.opts.platform,
+        this.opts.arch
+      ]))
+      .then(() => {
+        if (this.opts.prune) {
+          return hooks.promisifyHooks(this.opts.afterPrune, [
+            this.originalResourcesAppDir,
+            this.opts.electronVersion,
+            this.opts.platform,
+            this.opts.arch
+          ])
+        } else {
+          return true
+        }
+      })
```

Comment:
I don't know whether afterPrune and afterCopy hooks should be run, since neither of those operations actually took place, and `this.originalResourcesAppDir` doesn't actually exist.

Reply:
Yeah, I didn't what those hooks were for... I've changed it to just return early.

Reply:
I don't think returning early is the right solution... we may have to adjust the hook signature so that it's either the path to the resources app directory, or the path to the `app.asar` file (which would be a breaking change).

@MarshallOfSound since you have dealt with a lot of code relating to the hooks, you might have some insight here.